nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/TreeCommentManager.java

```diff
@@ -83,6 +85,18 @@
 
     public static final String SERVICE_WITHOUT_IMPLEMENTATION_MESSAGE = "This service implementation does not implement deprecated API.";
 
+    /**
+     * Counts how many comments where made on a specific document.
+     */
+    protected static final String QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_ECM_ANCESTOR = //
+            "SELECT " + ECM_UUID + " FROM Comment WHERE " + ECM_ANCESTORID + " = '%s'";
+
+    /**
+     * Counts how many comments where made by a specific user on a specific document.
+     */
+    protected static final String QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_ECM_ANCESTOR_AND_AUTHOR = //
+            QUERY_GET_COMMENTS_COUNT_FOR_DOCUMENT_BY_ECM_ANCESTOR + " AND " + COMMENT_AUTHOR + " = '%s'";
+
```

Comment:
we can use a buildQuery method i think there no need to extract them as a field class

Reply:
same as https://github.com/nuxeo/nuxeo/pull/3663#discussion_r368043049